lib/ansible/utils/encrypt.py

```diff
@@ -91,7 +92,14 @@ def hash(self, secret, salt=None, salt_size=None, rounds=None):
 
     def _salt(self, salt, salt_size):
         salt_size = salt_size or self.algo_data.salt_size
-        return salt or random_salt(salt_size)
+        ret = salt or random_salt(salt_size)
+        if re.search(r'[^./0-9A-Za-z]', ret):
```

Comment:
The algorithms above, all have this specific salt restriction.  I didn't validate every algo, but I don't see an issue with this.